### DISCUSSION TITLE
k8-service not showing metrics for cluster having multiple instance

### DISCUSSION THREAD

**Discussion by trs91km** [apache/skywalking#8291] [Q&A]
I have a kubernetes  which has 1 master and three worker nodes, when I try to enable monitoring for this cluster I am getting metrics only for k8-cluster, k8-nodes. k8-service not giving any metrics.

I replicate the same setup in kubernetes cluster having single node, I can see metrics of all the three k8-cluster, k8-nodes and k8-service.

Is this known issue or do I miss anything in configuration.

I have given enough permission for OAP instance in RBAC.

**Comment by wu-sheng** [comment]
I think I am not aware this case. The metrics of k8s cluster is collected by OpenTelemetry Collector, have you checked there first?

**Reply by wu-sheng** [reply]
cc @wankai123

**Reply by trs91km** [reply]
In OpenTelemetry Collector, I don't see any errors in its logs, it looks good. When I had this issue, Ichecked skywalking demo where I saw only single instance in that cluster so that thought of replicate the setup in single node k8 cluster and it worked well in it.

**Reply by wu-sheng** [reply]
No error doesn't mean it works well I am afraid. You need to check the metrics it collected.

**Reply by wankai123** [reply]
there is no additional configure for multi-nodes. I think skywalking demo has only one node just to save resources, it has multi-nodes before.

**Reply by trs91km** [reply]
I repeated the same with skwalking-showcase, facing same issue in multinode cluster, single node cluster working fine

**Reply by wu-sheng** [reply]
Again, recommend to checking metrics of OpenTelemetry collector. We are just processing it, and nothing difference between how many nodes of k8s, as we even don't talk to them.
